### PR TITLE
Fix macos again

### PR DESCRIPTION
**Pull request by SkySkimmer**
This reverts commit df115fbcdbd81387cd1211b556151a338c744eec (#19856) where the `brew upgrade` seems broken.

### CODE REVIEW

**Comment by SkySkimmer** [comment]
cc @mseri

File: .github/workflows/ci-macos.yml [hunk 1]
```diff
@@ -12,16 +12,12 @@ jobs:
...
-          brew install pkgconf
-          # work around https://github.com/actions/runner-images/issues/10984
-
       - name: Install system dependencies
         run: |
-          brew install gnu-time opam gtksourceview3 adwaita-icon-theme expat libxml2
+          brew unlink pkg-config
+          brew install gnu-time opam gtksourceview3 adwaita-icon-theme expat libxml2 pkgconf
+          brew unlink pkgconf
+          brew link pkg-config
```
Comment (mseri) [inline .github/workflows/ci-macos.yml:20]:
```suggestion
          brew install gnu-time opam gtksourceview3 adwaita-icon-theme expat libxml2
```
This should no longer be needed, the workers have been updated. In the logs you see it already:
```
 brew unlink pkg-config
  brew install gnu-time opam gtksourceview3 adwaita-icon-theme expat libxml2 pkgconf
  brew unlink pkgconf
  brew link pkg-config
  shell: /bin/bash -e {0}
Unlinking /usr/local/Cellar/pkgconf/2.3.0_1... 15 symlinks removed.
Warning: pkgconf 2.3.0_1 is already installed, it's just not linked.
To link this version, run:
  brew link pkgconf
```

Comment (SkySkimmer) [reply .github/workflows/ci-macos.yml]: we can try

**Comment by SkySkimmer** [comment]
Seems to work @coq/ci-maintainers

**Review by proux01 — APPROVED** [review]

**Comment by proux01** [comment]
@coqbot merge now

**Pull request merged**